src/FNA3D_Driver_Vulkan.c

```diff
@@ -2490,6 +2583,28 @@ static uint8_t VULKAN_INTERNAL_DeterminePhysicalDevice(
 		&renderer->memoryProperties
 	);
 
+	deviceLocalHeapSize = 0;
+	for (i = 0; i < renderer->memoryProperties.memoryHeapCount; i += 1)
+	{
+		if (	renderer->memoryProperties.memoryHeaps[i].flags &
+			VK_MEMORY_HEAP_DEVICE_LOCAL_BIT	)
+		{
+			if (renderer->memoryProperties.memoryHeaps[i].size > deviceLocalHeapSize)
+			{
+				deviceLocalHeapSize = renderer->memoryProperties.memoryHeaps[i].size;
+			}
+		}
+	}
+
+	deviceLocalHeapUsageFactorStr = SDL_GetHint("FNA3D_VULKAN_DEVICE_LOCAL_HEAP_USAGE_FACTOR");
+	if (deviceLocalHeapUsageFactorStr != NULL)
+	{
+		deviceLocalHeapUsageFactor = SDL_max(0, SDL_min(1.0, SDL_atof(deviceLocalHeapUsageFactorStr)));
```

Comment:
We may want to have a default for SDL_atof to avoid it accidentally passing 0 when the env var isn't a valid float:
```suggestion
		double factor = SDL_atof(deviceLocalHeapUsageFactorStr);
		if (factor > 0.0 && factor < 1.0)
		{
			deviceLocalHeapUsageFactor = factor;
		}
```